### ISSUE TITLE
org_ditaa_jar_path only configured by first emacs plugin update.

### ISSUE THREAD

**Issue by myme5261314** [purcell/emacs.d#338]
Hi, I'm a newbie to Emacs and just use this configuration first time today.

I'm facing a weird problem. I think it's better to report it if it's kind of bug of the configuration.

The situation is like this, after I cloned the configuration repo, and start Emacs the first time, it just downloaded many packages, after downloading finished. It's ok to open `org-mode` with <kbd>M-x org-mode</kbd>.
And the output of <kbd>C-h v org_ditaa_jar_path</kbd> is like below:
```
org-ditaa-jar-path's value is
"/home/xxx/.emacs.d/ditaa0_9.jar"
Original value was 
"/usr/share/emacs/24.3/lisp/contrib/scripts/ditaa.jar"
```
And then, just close Emacs and restart it again. It failed to open `org-mode` with the following `org_ditaa_jar_path` variable:
```
org-ditaa-jar-option's value is "-jar"

Documentation:
Option for the ditaa jar file.
Do not leave leading or trailing spaces in this string.
```
The error in the status line is `Symbols value for variable is void: org-ditaa-jar-path`.
And then every time you start Emacs, just the same, can't enter `org-mode`.

In conclusion, only my first run of Emacs would succeed to org-mode. The following sessions will fail to enter the org-mode.

I've reproduced this issue about 3 times.
OS: Ubuntu 14.04.3
Emacs: 24.3.1

**Comment by purcell** [comment]
Thanks for the report, but I can't duplicate the problem here with a recent version of Emacs. The code which refers to `org-ditaa-jar-path` only executes after `ob-ditaa` is loaded, and that guarantees that the variable is defined. So I can't tell from the information you provided what the problem might be -- perhaps a different Org version? Which version of Emacs are you using, and on which platform?

**Comment by myme5261314** [comment]
OK. I'm on Ubuntu 14.04.3, using emacs installed via *sudo apt-get install emacs*. I haven't do something special to org. Just the default version.

发自我的 iPhone

**Comment by purcell** [comment]
What does "M-x emacs-version" report?

**Comment by myme5261314** [comment]
Sorry, I've used self compiled emacs v24.5 now for `spacemacs` now.
The version by ubuntu default one is uninstalled.
I can only give you the version info from `apt-cache policy emacs24`
```
24.3+1-2ubuntu1 0
        500 http://mirrors.ustc.edu.cn/ubuntu/ trusty/main amd64 Packages
```

**Comment by purcell** [comment]
Yeah, really hard for me to reproduce this, sorry. There's not really a set of steps I can follow, and the config works fine for me with various emacs versions.

**Comment by jiaotang888** [comment]
![01](https://cloud.githubusercontent.com/assets/13150308/14481683/ef7385ee-0167-11e6-80b8-dd2e3c44093a.jpg)
![02](https://cloud.githubusercontent.com/assets/13150308/14481684/efc9e498-0167-11e6-9af3-fb5273e1d7ea.jpg)

**Comment by ryc111** [comment]
Same problem, ubuntu, emacs 24.3.1, fixed by adding setq org-ditaa-jar-path...

**Comment by purcell** [comment]
Thanks -- I tracked this down to an oddity of the org-mode in Emacs 24.3, whereby it didn't actually define org-ditaa-jar-path, despite using it. Fixed in master.

**Comment by redguardtoo** [comment]
24.3 doesn't have ob-ditta.el, either use 24.4 or install latest org-mode.

**Comment by purcell** [comment]
My Emacs 24.3 *definitely* has `ob-ditaa.el`.

**Comment by redguardtoo** [comment]
line 48 at https://github.com/tkf/org-mode/blob/master/lisp/ob-ditaa.el 
maybe it's different versions of ob-ditta.el

**Comment by purcell** [comment]
Yep, the version bundled in 24.3 is this one, which is different: http://git.savannah.gnu.org/cgit/emacs.git/tree/lisp/org/ob-ditaa.el?id=emacs-24.3
